extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -48,73 +49,96 @@
   private static final String WORK_ITEM_FILE_EXTENSION = ".json";
   private static final String DEFAULT_CONTENT_FOLDER = "system/default-content";
   private String environmentVariablesFolder;
-
+  private String ERROR = "error";
+  private String OK = "ok";
+  private String WI_STATUS = "/wi-status.";
   //for unit testability
   String getSolutionPath( ) {
     return PentahoSystem.getApplicationContext().getSolutionPath( DEFAULT_CONTENT_FOLDER );
   }
 
+  //places a failure breadcrumb in the specified folder
+  private void placeBreadcrumbFile( String folderPath, boolean success ) {
+    String ext = ERROR;
+    if ( success ) {
+      ext = OK;
+    }
+    String breadcrumbFilename = folderPath + WI_STATUS + ext;
+    File breadcrumb = new File( breadcrumbFilename );
+    try {
+      if ( !breadcrumb.createNewFile() ) {
+        logger.error( Messages.getInstance().getErrorString( "ActionInvokerSystemListener.ERROR_0004_COULD_NOT_WRITE_STATUS" ) );
+      }
+    } catch ( IOException e ) {
+      logger.error( Messages.getInstance().getErrorString( "ActionInvokerSystemListener.ERROR_0004_COULD_NOT_WRITE_STATUS" ) );
+    }
+  }
   /**
    * Runs Work Items described in json files upon startup of the Pentaho Server.
-   * @param session
-   * @return
+   *
+   * @param session The Pentaho Session
```

Comment:
it's good practice to state the object type of the parameter and link to it - the javadoc has proper hyper links: {@link IPentahoSession}